### PR TITLE
Add go example

### PR DESCRIPTION
**Pull request by jackkleeman**

### CODE REVIEW

File: .github/workflows/pre-release.yml
Comment (slinkydeveloper) [inline .github/workflows/pre-release.yml:1]: You forgot to update the commit message string.

Comment (slinkydeveloper) [reply .github/workflows/pre-release.yml:1]: And to change the `add-paths` at the bottom.

File: templates/go/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+COPY go.* ./
+RUN go mod download
+COPY . ./
+
+RUN CGO_ENABLED=0 go build -v -o server
+
+FROM alpine
+COPY --from=builder /app/server /app/server
+
+CMD ["/app/server"]
```
[the remark this answers is not in the document]
Comment (jackkleeman) [reply templates/go/Dockerfile:14]: idk, people can use that if they want but probably a dockerfile is what people are most likely to expect in a template?

[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply templates/go/Dockerfile:14]: > but probably a dockerfile is what people are most likely to expect in a template?

I can't tell if they would expect a Dockerfile, I think it's sufficient to give people one way to build a container image. The reason I suggested ko is that is so much common those days, it's used even in some of the cncf build pipelines, and it's literally one command to write down in the README. But if you wanna go with the dockerfile, i'm ok with it.

**Pull request merged**